src/ui/windows/TogglDesktop/TogglDesktop/ui/controls/MiniTimer.xaml.cs

```diff
@@ -184,114 +104,6 @@ protected override void OnGotKeyboardFocus(KeyboardFocusChangedEventArgs e)
             {
                 this.descriptionTextBox.Focus();
             }
-            else
-            {
-                this.startStopButton.Focus();
-            }
-        }
-
-        #endregion
-
-        #region controlling
-
-        private void startStop()
-        {
-            this.acceptNextUpdate = true;
-
-            if (this.isRunning)
-            {
-                this.stop();
-            }
-            else
-            {
-                this.start();
-            }
-        }
-
-        private void tryOpenEditViewIfRunning(MouseButtonEventArgs e, string focusedField)
-        {
-            if (this.isRunning)
-            {
-                if (e.ClickCount == 2)
-                {
-                    using (Performance.Measure("opening edit view from timer, focussing " + focusedField))
-                    {
-                        Toggl.Edit(this.runningTimeEntry.GUID, false, focusedField);
-                    }
-
-                    e.Handled = true;
-                }
-            }
-        }
-
-        private void start()
-        {
-            var durationString = durationTextBox.Text;
-            using (Performance.Measure("starting time entry from timer"))
-            {
-                var guid = Toggl.Start(
-                    this.descriptionTextBox.Text,
-                    "",
-                    completedProject.TaskID,
-                    completedProject.ProjectID,
-                    "",
-                    completedProject.Tags,
-                    IsMiniTimer
-                );
-
-                if (!string.IsNullOrEmpty(guid) && !string.IsNullOrEmpty(durationString))
-                {
-                    Toggl.SetTimeEntryDuration(guid, durationString);

```

Comment:
the functionality for setting starting duration from the mini timer has disappeared after this refactoring

Reply:
Is there any sense, that this is present only in MiniTimer, but not present in Timer?

Reply:
It's temporarily missing from the main Timer's design. It will probably be added to the Timer later.

Reply:
This functionality is still not working